app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala

```diff
@@ -833,31 +871,31 @@ class SectionRenderingService(
               upscanInitiate
             )
           case FileUpload(fileUploadProvider) =>
-            val fileId: FileId =
-              ei.envelope
-                .find(formComponent.modelComponentId)
-                .map(_.fileId)
-                .getOrElse(ei.envelope.mapping.fileIdFor(formComponent.id))
-
+            val fileId: FileId = ei.envelope
+              .find(formComponent.modelComponentId)
+              .map(_.fileId)
+              .getOrElse(ei.envelope.mapping.fileIdFor(formComponent.id))
             fileUploadProvider match {
               case FileUploadProvider.Upscan =>
-                val upscanInitiateResponse: UpscanInitiateResponse = upscanInitiate.get(formComponent.id)
-                val hiddenFields: List[Html] = upscanInitiateResponse.uploadRequest.fields.toList.map {
-                  case (name, value) =>
-                    html.form.snippets.hidden(name, value)
-                }
                 val fileUploadName = "file"
-                val attributes = Map("upscan" -> "upscan")
+                val otherAttributes =
+                  if (upscanData.isDefined) Map("form" -> upscanData.get.formMetaData.htmlId)
+                  else Map.empty[String, String]
+
+                val formActionUrl = upscanData.map(_.url).getOrElse("")
```

Comment:
Calling `.get` on value of `Option` type is bad and please don't do it.

In this case when there is no `upscanData` available what this implementation would result in is a broken form. 

Right thing to do here is to enforce presence of `upscanData`, in case `upscanData` are not available there is bug in our code and there is nothing better we can do than to throw and exception.

```
case FileUploadProvider.Upscan =>
  upscanData match {
    case None => throw new IllegalArgumentException(TODO provide nice error message with as much info as possible)
    case Some(upscanData) =>
      val fileUploadName = "file"
      ...
  }